File: csharp/AveroxCrypto.cs
/**
 * salman 40 - Enterprise Cryptographic SDK for .NET
 * Generated: 2025-08-28T10:28:26.241Z
 */

using System;
using System.Security.Cryptography;
using System.Text;

namespace Averox.Crypto
{
    public class AveroxCrypto : IDisposable
    {
        private readonly byte[] masterKey;
        private bool disposed = false;
        
        public AveroxCrypto(byte[] masterKey)
        {
            if (masterKey == null || masterKey.Length < 32)
                throw new ArgumentException("Master key must be at least 32 bytes");
            
            this.masterKey = new byte[masterKey.Length];
            Array.Copy(masterKey, this.masterKey, masterKey.Length);
        }
        
        public EncryptedData Encrypt(string plaintext, byte[] aad = null)
        {
            using (var aes = Aes.Create())
            {
                aes.Key = DeriveKey();
                aes.Mode = CipherMode.GCM;
                
                var iv = new byte[12];
                RandomNumberGenerator.Fill(iv);
                aes.IV = iv;
                
                var plaintextBytes = Encoding.UTF8.GetBytes(plaintext);
                var ciphertext = new byte[plaintextBytes.Length];
                var tag = new byte[16];
                
                using (var encryptor = aes.CreateEncryptor())
                {
                    ((AesGcm)encryptor).Encrypt(iv, plaintextBytes, ciphertext, tag, aad);
                }
                
                return new EncryptedData
                {
                    IV = Convert.ToBase64String(iv),
                    Ciphertext = Convert.ToBase64String(ciphertext),
                    Tag = Convert.ToBase64String(tag)
                };
            }
        }
        
        private byte[] DeriveKey()
        {
            using (var pbkdf2 = new Rfc2898DeriveBytes(masterKey, Encoding.UTF8.GetBytes("averox-salt"), 100000, HashAlgorithmName.SHA256))
            {
                return pbkdf2.GetBytes(32);
            }
        }
        
        public void Dispose()
        {
            if (!disposed)
            {
                Array.Clear(masterKey, 0, masterKey.Length);
                disposed = true;
            }
        }
    }
    
    public class EncryptedData
    {
        public string IV { get; set; }
        public string Ciphertext { get; set; }
        public string Tag { get; set; }
    }
}

File: dart/lib/averox_crypto.dart
/**
 * salman 40 - Enterprise Cryptographic SDK for Dart/Flutter
 * Generated: 2025-08-28T10:26:18.964Z
 */

import 'dart:convert';
import 'dart:typed_data';
import 'package:crypto/crypto.dart';
import 'package:pointycastle/export.dart';

class AveroxCrypto {
  final Uint8List _masterKey;
  
  AveroxCrypto(this._masterKey) {
    if (_masterKey.length < 32) {
      throw ArgumentError('Master key must be at least 32 bytes');
    }
  }
  
  EncryptedData encrypt(String plaintext, {Uint8List? aad}) {
    final key = _deriveKey();
    final iv = _generateRandomBytes(12);
    
    final cipher = GCMBlockCipher(AESEngine());
    final params = AEADParameters(KeyParameter(key), 128, iv, aad);
    cipher.init(true, params);
    
    final plaintextBytes = utf8.encode(plaintext);
    final ciphertext = Uint8List(plaintextBytes.length + 16);
    final len = cipher.processBytes(plaintextBytes, 0, plaintextBytes.length, ciphertext, 0);
    cipher.doFinal(ciphertext, len);
    
    return EncryptedData(
      iv: base64.encode(iv),
      ciphertext: base64.encode(ciphertext.sublist(0, plaintextBytes.length)),
      tag: base64.encode(ciphertext.sublist(plaintextBytes.length))
    );
  }
  
  Uint8List _deriveKey() {
    final pbkdf2 = PBKDF2KeyDerivator(HMac(SHA256Digest(), 64));
    pbkdf2.init(Pbkdf2Parameters(utf8.encode('averox-salt'), 100000, 32));
    return pbkdf2.process(_masterKey);
  }
  
  Uint8List _generateRandomBytes(int length) {
    final random = SecureRandom('Fortuna');
    final seed = Uint8List(32);
    for (int i = 0; i < 32; i++) {
      seed[i] = (DateTime.now().millisecondsSinceEpoch + i) & 0xFF;
    }
    random.seed(KeyParameter(seed));
    return random.nextBytes(length);
  }
}

class EncryptedData {
  final String iv;
  final String ciphertext;
  final String tag;
  
  EncryptedData({required this.iv, required this.ciphertext, required this.tag});
}